test/Sema/access-level-import-inconsistent-same-file.swift

```diff
@@ -0,0 +1,105 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+/// Build the library.
+// RUN: %target-swift-frontend -emit-module %t/Lib.swift -module-name Lib1 -o %t
+// RUN: %target-swift-frontend -emit-module %t/Lib.swift -module-name Lib2 -o %t
+// RUN: %target-swift-frontend -emit-module %t/Lib.swift -module-name Lib3 -o %t
+// RUN: %target-swift-frontend -emit-module %t/Lib.swift -module-name Lib4 -o %t
+// RUN: %target-swift-frontend -emit-module %t/Lib.swift -module-name Lib5 -o %t
+
+/// Test main cases.
+// RUN: %target-swift-frontend -typecheck -verify %t/Client.swift -I %t
+// RUN: %target-swift-frontend -typecheck -verify %t/Client_Scoped.swift -I %t
+// RUN: %target-swift-frontend -typecheck -verify %t/Client_Clang.swift -I %t
+
+/// Test language mode specific variants.
+// RUN: %target-swift-frontend -typecheck -verify %t/Client_Swift6.swift -I %t \
+// RUN:   -enable-upcoming-feature InternalImportsByDefault
+// RUN: %target-swift-frontend -typecheck -verify %t/Client_Swift5.swift -I %t \
+// RUN:   -swift-version 5
+
+//--- Lib.swift
+public struct Type1 {}
+public struct Type2 {}
+
+//--- Client.swift
+/// Simple public vs internal.
+public import Lib1 // expected-note {{imported 'public' here}}
+// expected-warning @-1 {{public import of 'Lib1' was not used in public declarations or inlinable code}}
```

Comment:
If this test is not specifically meaning to exercise the `was not used in public declarations` diagnostic, I would find it easier to follow if this diagnostic were suppressed by some trivial use of declarations from the library.

Reply:
I followed your suggestion and all of these messages are gone, except one which is a bug I'll have to fix in the other diagnostic logic.